Content/posts/async-template-updates.md

```diff
@@ -0,0 +1,32 @@
+---
+date: 2023-04-30 12:00
+description: You may notice some changes in Vapor's templates - we've updated then for Swift 5.8, with a unified target and using the latest `@main` syntax.
+tags: templates
+authors: Tim
+authorImageURLs: /author-images/tim.jpg
+---
+# Updating Vapor's Templates for Swift 5.8
+
+If you've run `vapor new` recently, you may have noticed some changes to the templates. We've updated them for Swift 5.8, with a unified target and using the latest `@main` syntax.
+
+## Swift 5.8 and a Unified Target
+
+The template now has a minimum supported Swift version of 5.8 to allow you to easily use the latest Swift features. The Dockerfile has been updated to use Swift 5.8 as well so deploying should be as easy as ever.
+
+Swift 5.3 (yes believe it or not it was that long ago) introduced the `@main` syntax for defining the entry point of your application. This allows you to annotate any type as the entry point of your application. Swift 5.5 finally allowed you to write tests for an `executableTarget` in a Swift Package, removing the need for a separate `Run` target as Vapor has traditionally used.
+
+We held off migrating to this new syntax partly for backwards compatibility, to avoid breaking all the tutorials, books and examples out there and because there wasn't a clear reason to migrate. Now that Vapor supports Swift Concurrency, we need to start planning on how to support asynchronous setup functions for your apps. This is the first step to do so as you can now have an `async` entrypoint.
```

Comment:
```suggestion
We held off migrating to this new syntax partly for backwards compatibility, to avoid breaking all the tutorials, books, and examples out there, and because there wasn't a clear reason to migrate. Now that Vapor supports Swift Concurrency, we need to start planning on how to support asynchronous setup functions and commands for your apps. This change takes the first step in that direction by moving to an `async` entry point.
```